source/extensions/tracers/xray/BUILD

```diff
@@ -13,20 +13,28 @@ envoy_package()
 envoy_cc_library(
     name = "xray_lib",
     srcs = [
+        "localized_sampling.cc",
+        "tracer.cc",
         "util.cc",
         "xray_tracer_impl.cc",
     ],
     hdrs = [
+        "localized_sampling.h",
+        "reservoir.h",
         "sampling_strategy.h",
         "tracer.h",
         "util.h",
         "xray_configuration.h",
         "xray_tracer_impl.h",
     ],
+    external_deps = [
+        "rapidjson",
```

Comment:
Unfortunately we're planning on deprecating rapidjson soon and (as I understand it) don't plan on providing a different JSON parser. It should be possible to load the JSON into a ProtobufWkt::Struct and access its contents from there, however.

cc @htuch and @mattklein123

Reply:
Yep, please use https://github.com/envoyproxy/envoy/blob/54169bbc33f122569df5193b01f2d2908924e931/source/common/protobuf/utility.cc#L135, this is the supported way to parse JSON going forward.

Reply:
I starting writing the protobuf message that matches the JSON schema for X-Ray and I ran into a snag. `default` is a protobuf keyword but it's also the name of a key in X-Ray's JSON schema:
```json
{
  "version": 2,
 "default": {
    "fixed_target": 1
  },
  "rules": [
    {
      "description": "X-Ray rule",
      "host": "*",
      "http_method": "*",
      "url_path": "/api/move/*",
      "fixed_target": 0,
      "rate": 0.05
    }
  ]
}
```
How do you normally get around that?

Reply:
I haven't seen this before; I didn't think that protobuf keywords get special treatment, there is supposed to be a 1:1 mapping between JSON and proto3 https://developers.google.com/protocol-buffers/docs/proto3#json

Reply:
> It should be possible to load the JSON into a ProtobufWkt::Struct and access its contents from there

Ugh, I'm dumb. I thought I needed to define a message that maps to the schema I'm expecting to be able to access the fields.

I get it now though. Thanks.

Reply:
I think having a protobuf message that describes the format would be helpful in that you could delete an awful lot of the validation code you have. But if the format's fixed and default isn't allowed as a field name, I guess that's out.

Reply:
>I think having a protobuf message that describes the format would be helpful in that you could delete an awful lot of the validation code you have. 

Right, that's what I was trying to do at first. It made a lot of sense.
>But if the format's fixed and default isn't allowed as a field name, I guess that's out.

Yeah my hands are tied there. That's the format used by X-Ray in all of its SDKs :/